src/Response/JsonApiResponse.php

```diff
@@ -3,74 +3,91 @@
 namespace Tomaj\NetteApi\Response;
 
 use DateTimeInterface;
-use Nette;
-use Nette\Application\Responses\JsonResponse;
+use Nette\Http\IRequest;
+use Nette\Http\IResponse;
+use Nette\SmartObject;
+use Nette\Utils\Json;
 
-class JsonApiResponse extends JsonResponse
+class JsonApiResponse implements ResponseInterface
 {
-    /**
-     * @var integer
-     */
+    use SmartObject;
+
+    /** @var integer */
     private $code;
 
-    /**
-     * @var string
-     */
+    /** @var mixed */
+    private $payload;
+
+    /** @var string */
+    private $contentType;
+
+    /** @var string */
     private $charset;
 
-    /**
-     * @var string|int|bool|DateTimeInterface
-     */
+    /** @var bool|DateTimeInterface|int|string */
     private $expiration;
 
     /**
-     * Create JsonApiResponse
-     * This class only wrap JsonResponse from Nette and add possibility
-     * to setup response code and automaticaly set content type
+     * This class is only copy of JsonResponse from Nette with added possibility
+     * to setup response code, content type, charset and expiration
      *
      * @param integer $code
      * @param mixed $data
      * @param string $contentType
      * @param string $charset
-     * @param string|int|bool|DateTimeInterface $expiration
+     * @param bool|DateTimeInterface|int|string $expiration
```

Comment:
fujky fujky

Reply:
uz ked to breakujeme, tak nespravime toto nejak standartne?

Reply:
no to som sa chcel dohodnut, toto je upgrade len aby to slo s novym nette (cize to nie je az taky BC Break ako ked pridame typy a pod)

ale chcem spravit aj return typy a vsetko typovane. Otazka je ci uz teraz alebo v dalsej major verzii

Reply:
ja by som to dal asi uz aj tu. zbytocne dalsi bc break by to bol. a rovnako aj tu sa menia nejake interfacy s typmi cize nevidim dovod to nedat rovno tu :-)

Reply:
hmm nemenia, resp. nechcel som menit :) mozno som sa niekde sekol :) ale nebolo to v plane

Reply:
som tam videl ze nieco sa uz inak dedi a su tam nejake typovane interfaci. ale to je jedno. vydame novu major verziu a potom to znovu breakneme a vydame dalsiu major verziu? mne sa zda lepsie to spojit do 1 :-)

Reply:
ano lebo JsonResponse je uz final takze sa od neho neda dedit, pridal som NetteApi ResponseInterface ktory rozsiruje Application/IResponse o metodu getCode, inak nic velke :) ale OK suhlasim, ked break tak poriadny :)